guides/source/active_record_basics.md

```diff
@@ -71,107 +70,201 @@ Convention over Configuration in Active Record
-row in that table with the attributes of the instances of your model. Suppose
-that the `products` table was created using an SQL (or one of its extensions) statement like:
+This will create a `Book` model, mapped to a `books` table in the
+database, where each column in the table is mapped to attributes of the `Book` class. An instance of `Book` can represent a row in the `books` table. The `books` table can be created using an SQL statement like this:
 
 ```sql
-CREATE TABLE products (
+CREATE TABLE books (
   id int(11) NOT NULL auto_increment,
-  name varchar(255),
+  title varchar(255),
+  author varchar(255),
   PRIMARY KEY  (id)
 );
 ```
 
-The schema above declares a table with two columns: `id` and `name`. Each row of
-this table represents a certain product with these two parameters. Thus, you
-would be able to write code like the following:
+Database tables in Rails are typically created using [Active Record Migrations](#migrations) and not raw SQL. A migration for the `books` table above can be generated like this:
+
+```bash
+$ rails generate migration CreateBooks title:string author:string
+```
+
+and results in this:
 
 ```ruby
-p = Product.new
-p.name = "Some Book"
-puts p.name # "Some Book"
+# Note:
+# The `id` column, as the primary key, is automatically created by convention.
+# Columns `created_at` and `updated_at` are added by `t.timestamps` line.
+
+# /db/migrate/20240220143807_create_books.rb
+class CreateBooks < ActiveRecord::Migration
+  def change
+    create_table :books do |t|
+      t.string :title
+      t.string :author
+
+      t.timestamps
+    end
+  end
+end
 ```
 
+The SQL as well as the migration above declare a table with three columns: `id`,
+`title`, and `author`. Each row of this table can be represented by an instance
+of the `Book` class with the same three attributes: `id`, `title`, and `author`.
+You can access a book's attributes like this:
+
+```irb
+irb> book = Book.new
+=> #<Book:0x00007fbdf5e9a038 id: nil, title: nil, author: nil, created_at: nil, updated_at: nil>
+
+irb> book.title = "The Hobbit"
+=> "The Hobbit"
+irb> book.title
+=> "The Hobbit"
+```
+
+NOTE: You can generate the Active Record model class as well as a matching migration with this command `bin/rails generate model Book title:string author:string`. This creates both `/app/models/book.rb` and `/db/migrate/20240220143807_create_books.rb` files.
+
+### Creating Namespaced Models
+
+Active Record models are placed under the `app/models` directory by default. But you may want to organize your models by placing similar models under their own folder and namespace. For example, `order.rb` and `review.rb` under `app/models/product` with `Product::Order` and `Product::Review` class names, respectively. You can create namespaced models with Active Record.
+
+In the case where `Product` module does not already exist, the `generate` command will create everything like this:
+
+```irb
+$ rails generate model Product/Order
```

Comment:
```suggestion
$ bin/rails generate model Product/Order
```

Reply:
Perhaps we should use `Product::Order` instead of `Product/Order` to make it more like a Ruby namespace? It would match the generated code too:

```suggestion
$ bin/rails generate model Product::Order
```